contracts/wrapper/NameWrapper.sol

```diff
@@ -456,21 +416,40 @@ contract NameWrapper is
      * @param _fuses initial fuses for the wrapped subdomain
      */
 
-    function setSubnodeRecordAndWrap(
+    function setSubnodeRecord(
         bytes32 parentNode,
         string calldata label,
         address newOwner,
         address resolver,
         uint64 ttl,
         uint96 _fuses
-    ) public override {
+    )
+        public
+        onlyTokenOwner(parentNode)
+        canCallSetSubnodeOwner(parentNode, keccak256(bytes(label)))
+    {
         bytes32 labelhash = keccak256(bytes(label));
         bytes32 node = _makeNode(parentNode, labelhash);
-        bytes memory name = _addLabel(label, names[parentNode]);
-
-        setSubnodeRecord(parentNode, labelhash, address(this), resolver, ttl);
 
-        _wrap(node, name, newOwner, _fuses);
+        if (ens.owner(node) != address(this)) {
+            ens.setSubnodeRecord(
```

Comment:
This can go outside the if statement, since it's the same in both branches.

Reply:
Can't do this, because it needs to go before wrap for re-entrancy, which I think we discussed today.